tests/util_test.js

```diff
@@ -34,5 +34,16 @@ describe('Utils', () => {
                     b: 2,
                 });
         });
+        it('does not mutate Arrays into Objects', () => {
```

Comment:
The arrays aren't being mutated, it's more that they're being replaced by objects, so I'm not sure I agree with this name.

For future readers though (who don't have context about what was happening in this bug) I'd prefer if we didn't call this "it doesn't..." and instead say "it does..." like "it replaces arrays instead of trying to merge them." How does that sound?